Write the base commit to .git/base_sha

aconrad opened the pull request:
This is useful for when we need to know the SHA-1 of the base branch when the build ran which isn't necessarily the HEAD of `master` (or whatever the base branch is).

jtarchie commented:
@aconrad, thanks for the PR.

Please run the tests locally, see the `README.md` under Testing for more information.

With features like this, please write a test, and support both the file and the `git config` option.

Cheers.

jtarchie commented:
@aconrad, are you able to provide the other requested changes? tests...

aconrad commented:
Hey sorry @jtarchie, been working on something else and haven't found the bandwidth yet. This PR started out raw but I wanted to make sure it was okay before spending too much time on it. I never worked with Ruby so it might take me a little bit to figure out what's going on / how to write tests. I just need to sit and take a look at it, haha!

aconrad commented:
@jtarchie let me know how this looks

jtarchie commented:
LGTM! I'll merge this in with some others later.

aconrad commented:
Also, happy to squash my commits if that's something you care about.

aconrad commented:
I noticed that the check fails. Anything for me to fix here?

jtarchie commented:
Run the tests locally, see if anything happens.

`docker build .` will run the tests.

aconrad commented:
Running `docker build .` I get a lot of output where things are being downloaded/installed, but then I get:

```
[... snip gem stuff being installed ...]
Bundle complete! 10 Gemfile dependencies, 46 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
........I, [2018-03-12T21:42:24.986003 #460]  INFO -- : puffing-billy: Proxy listening on http://localhost:43679
.Cloning into '/tmp/d20180312-460-f0w86h'...
Switched to branch 'pr-foo'
Fetching pr-foo
................fatal: repository 'invalid_git_uri' does not exist
.Cloning into '/tmp/d20180312-460-1o7w5f2'...
Switched to branch 'pr-foo'
Fetching pr-foo
.Cloning into '/tmp/d20180312-460-fubd3r'...
Switched to branch 'pr-foo'
Fetching pr-foo
.Cloning into '/tmp/d20180312-460-1tdvfxn'...
Switched to branch 'pr-foo'
Fetching pr-foo
.Cloning into '/tmp/d20180312-460-1kei54x'...
Switched to branch 'pr-foo'
Fetching pr-foo
..Cloning into '/tmp/d20180312-460-10jmvzu'...
Switched to branch 'pr-foo'
Fetching pr-foo
.Cloning into '/tmp/d20180312-460-1bt4tbt'...
Switched to branch 'pr-foo'
.Cloning into '/tmp/d20180312-460-1dy2wo5'...
Switched to branch 'pr-foo'
Fetching pr-foo
.Cloning into '/tmp/d20180312-460-etwh3s'...
Switched to branch 'pr-foo'
Fetching pr-foo
.Cloning into '/tmp/d20180312-460-ehrikg'...
Switched to branch 'pr-foo'
Fetching pr-foo
.Cloning into '/tmp/d20180312-460-901u9s'...
Switched to branch 'pr-foo'
error: pathspec 'path1' did not match any file(s) known to git.
error: pathspec 'path2' did not match any file(s) known to git.
Fetching pr-foo
.Cloning into '/tmp/d20180312-460-1331i6e'...
Switched to branch 'pr-foo'
error: pathspec 'path1' did not match any file(s) known to git.
error: pathspec 'path2' did not match any file(s) known to git.
Fetching pr-foo
..................................I, [2018-03-12T21:42:28.043450 #460]  INFO -- : puffing-billy: Proxy listening on http://localhost:45957
I, [2018-03-12T21:42:29.097245 #460]  INFO -- : puffing-billy: STUB GET for 'https://test.example.com:443/repos/jtarchie/test/pulls?direction=asc&per_page=100&sort=updated&state=open'
.I, [2018-03-12T21:42:29.113867 #460]  INFO -- : puffing-billy: Proxy listening on http://localhost:41655
I, [2018-03-12T21:42:29.951257 #460]  INFO -- : puffing-billy: STUB GET for 'https://api.github.com:443/repos/jtarchie/test/pulls/1'
.I, [2018-03-12T21:42:30.358435 #460]  INFO -- : puffing-billy: Proxy listening on http://localhost:35977
I, [2018-03-12T21:42:31.364666 #460]  INFO -- : puffing-billy: STUB POST for 'https://api.github.com:443/repos/jtarchie/test/statuses/43f1cd3c275d6aa0d1cce52d293ba81f2c8c9b5c'
.

Finished in 6.53 seconds (files took 0.88572 seconds to load)
74 examples, 0 failures

Removing intermediate container 6211a8ef9a7b
 ---> b0f4f898c279
Step 19/19 : FROM resource
 ---> 91c4cadbd690
Successfully built 91c4cadbd690
```

It _seems_ happy but the output of the tests seems to have errors in it. Doing the same against master results in pretty much the same output, only running 72 tests instead of 74.
